lib/python/flame/mode/horizontal/syncfl/fwdllm_aggregator.py

```diff
@@ -1323,6 +1391,131 @@ def _aggregate_grads_sync(self, tag: str) -> None:
                 )
                 break
 
+        # second loop to poll more if needed
+        num_freed = 0 # if at least one is freed, exit loop
+        while self._agg_goal_cnt < self._agg_goal and not self.ends_not_selected_yet: 
+            for msg, metadata in channel.recv_fifo(channel.ends(), 1):
```

Comment:
what will happen if there are no trainers left to get updates from? will it block indefinitely or will it exit?